### ISSUE TITLE
Optional rule on key in Multi-Dimensional array not functioning as expected

### ISSUE THREAD

**Issue by BenWalters** [vlucas/valitron#279]
Hi,

I'm trying to apply the 'optional' rule on a key in a multi-dimensional array:

```
$params = [
        'level2' => array(
            [
                'level3_1' => 'value',
                'level3_2' => 'value'
            ],
            [
                'level3_1' => 'value',
                'level3_2' => 'value',
                'level3_optional' => true,
            ]
        )
]
```

I've tried using the following rule:
```
        $v->rule('optional', ['level2.*.level3_optional']);
        $v->rule('boolean', ['level2.*.level3_optional']);
```

And also attempted to implement it in the following way:
```
$v->rules([
    'boolean' => [
        ['level2.*.level3_optional']
    ],
    'optional' => [
        ['level2.*.level3_optional']
    ]
]);
```

Both return:
```
Array
(
    [level3_optional] => Array
        (
            [0] => Level3_optional must be a boolean
        )

)
```

This is still the case even if the 'level3_optional' key doesn't exist at all.
Is this a bug or am I doing something incorrectly?

**Comment by willemwollebrants** [comment]
Are you using the latest version (v1.4.4)? This was a bug before but it should have been fixed in this PR: #263

**Comment by BenWalters** [comment]
@willemwollebrants, my bad I had v1.4.3! Thanks for the pointer!
